File: kickstart/2022/Coding Practive with Kick Start Session #1/centauri_prime.c
#include <stdio.h>
#include <string.h>

char* GetRuler(char* kingdom) {
    // TODO: implement this method to determine the ruler name, given the kingdom.
    char* ruler = "";
    char* vowels = "AEIOUaeiou";
    char* ys = "Yy";
    char last;

    last = kingdom[strlen(kingdom)-1];
    if (strchr(vowels, last) != NULL) {
        ruler = "Alice";
    } else if (strchr(ys, last) != NULL) {
        ruler = "nobody";
    } else {
        ruler = "Bob";
    }

    return ruler;
}

int main() {
    int testcases;
    scanf("%d", &testcases);
    char kingdom[100];

    for (int t = 1; t <= testcases; ++t) {
            scanf("%s", &kingdom);
            printf("Case #%d: %s is ruled by %s.\n", t, kingdom, GetRuler(kingdom));
    }
    return 0;
}

File: kickstart/2022/Coding Practive with Kick Start Session #1/centauri_prime.py
# TODO: Complete the get_ruler function
def get_ruler(kingdom):
    ruler = ''
    last = kingdom[-1]
    vowels =  ['A', 'E', 'I', 'O', 'U', 'a', 'e', 'i', 'o', 'u']
    ys = ['Y', 'y']
    if last in vowels:
        ruler = 'Alice'
    elif last in ys:
        ruler = 'nobody'
    else:
        ruler = 'Bob'
    return ruler

def main():
    # Get the number of test cases
    t = int(input())
    for case in range(1, t+1):
        # Get the kingdom
        kingdom = input()
        print('Case #{}: {} is ruled by {}.'.format(case, kingdom, get_ruler(kingdom)))

if __name__ == '__main__':
  main()
